### PR TITLE
Release 2023.09.0

### LINKED ISSUE #8223: release 2023.09.0
### Is your feature request related to a problem?

We've accumulated a nice number of changes.

Can someone volunteer to do a release in the next few days?

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kmuehlbauer**
As requested in #8223 I'm preparing for Xarray release 2023.09.0.

ToDo: 

- [x] Add final date
- [x] Approve headline and other changes in `whats-new.rst`
- [x] add plan-to-merge PR

- [x] Closes #8223 
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
I'm planning to merge this tomorrow by 8 UTC and proceed. Please review the release headline and suggest any changes you might find useful.

File: doc/whats-new.rst [hunk 1]
```diff
@@ -31,8 +45,7 @@ New Features
 - Improved static typing of reduction methods (:pull:`6746`).
```
Comment (headtr1ck) [inline doc/whats-new.rst:45]: That's more a Bugfix than a new feature.

File: doc/whats-new.rst [hunk 2]
```diff
@@ -14,10 +14,24 @@ What's New
...
+Thanks to the 24 contributors to this release: Alexander Fischer, Amrest Chinkamol, Benoit Bovy, Darsh Ranjan, Deepak Cherian,
+Gianfranco Costamagna, Gregorio L. Trevisan, Illviljan, Joe Hamman, JR, Justus Magin, Kai Mühlbauer, Kian-Meng Ang, Kyle Sunden,
+Martin Raspaud, Mathias Hauser, Mattia Almansi, Maximilian Roos, András Gunyhó, Michael Niklas, Richard Kleijn, Riulinchen,
+Tom Nicholas, Wiktor Kraśnicki.
+
+We welcome the following new contributors to Xarray!: Alexander Fischer, Darsh Ranjan, Gianfranco Costamagna, Gregorio L. Trevisan,
+Kian-Meng Ang and Wiktor Kraśnicki.
+
+
+We welcome
```
Comment (headtr1ck) [inline doc/whats-new.rst:34]:
```suggestion
```

**Review by headtr1ck — APPROVED** [review]

**Review by benbovy — APPROVED** [review]
LGTM, thanks @kmuehlbauer.

**Review by mathause — APPROVED** [review]
Could remove the empty `Documentation` section.

File: doc/whats-new.rst [hunk 3]
```diff
@@ -14,10 +14,24 @@ What's New
...
+This release continues work on the new :py:class:`xarray.Coordinates` object, allows to provide `preferred_chunks` when
+reading from netcdf files, enables :py:func:`xarray.apply_ufunc` to handle missing core dimensions and fixes several bugs.
+
+Thanks to the 24 contributors to this release: Alexander Fischer, Amrest Chinkamol, Benoit Bovy, Darsh Ranjan, Deepak Cherian,
+Gianfranco Costamagna, Gregorio L. Trevisan, Illviljan, Joe Hamman, JR, Justus Magin, Kai Mühlbauer, Kian-Meng Ang, Kyle Sunden,
+Martin Raspaud, Mathias Hauser, Mattia Almansi, Maximilian Roos, András Gunyhó, Michael Niklas, Richard Kleijn, Riulinchen,
+Tom Nicholas, Wiktor Kraśnicki.
+
+We welcome the following new contributors to Xarray!: Alexander Fischer, Darsh Ranjan, Gianfranco Costamagna, Gregorio L. Trevisan,
+Kian-Meng Ang and Wiktor Kraśnicki.
```
Comment (keewis) [inline doc/whats-new.rst:31]: Should the authors of #7578 and #7955 be included here?

Comment (kmuehlbauer) [reply doc/whats-new.rst:31]: Yes, sure. Thanks for catching this, @keewis.

Comment (kmuehlbauer) [reply doc/whats-new.rst]: I've added #7955 to the docs section, it had no entry in whats-new.rst.

**Comment by kmuehlbauer** [comment]
Upstream issues are not related to any changes in this release, afaict. Going to proceed according the announced schedule.

**Comment by kmuehlbauer** [comment]
Thanks all! :+1:

**Pull request merged**